interfaces/cython/cantera/ctml2yaml.py

```diff
@@ -0,0 +1,2188 @@
+            Tmin = float(node.get("Tmin", 0))
+            Tmax = float(node.get("Tmax", 0))
+            if not Tmin or not Tmax:
+                raise MissingXMLAttribute("Tmin and Tmax must both be specified", node)
+            temperature_ranges.add(Tmin)
+            temperature_ranges.add(Tmax)
+            float_array = node.find("floatArray")
+            if float_array is None:
+                raise MissingXMLNode(
+                    "{} entry missing floatArray node".format(poly_type), node
+                )
+            unsorted_data[Tmin] = FlowList(
+                map(float, clean_node_text(float_array).split(","))
+            )
+
+        if len(temperature_ranges) != len(model_nodes) + 1:
+            raise ValueError(
+                "The midpoint temperature is not consistent between {} "
+                "entries".format(poly_type)
+            )
+        data = []
+        for key in sorted(unsorted_data.keys()):
+            data.append(unsorted_data[key])
+
+        return data, FlowList(sorted(temperature_ranges))
+
+    def Shomate(
+        self, thermo: etree.Element
+    ) -> Dict[str, Union[str, "THERMO_POLY_TYPE"]]:
+        """Process a Shomate polynomial from XML to a dictionary."""
+        thermo_attribs = BlockMap({"model": "Shomate"})
+        data, temperature_ranges = self.process_polynomial(thermo, "Shomate")
+        thermo_attribs["data"] = data
+        thermo_attribs["temperature-ranges"] = temperature_ranges
+        return thermo_attribs
+
+    def NASA(self, thermo: etree.Element) -> Dict[str, Union[str, "THERMO_POLY_TYPE"]]:
+        """Process a NASA 7 thermo entry from XML to a dictionary."""
+        thermo_attribs = BlockMap({"model": "NASA7"})
+        data, temperature_ranges = self.process_polynomial(thermo, "NASA")
+        thermo_attribs["data"] = data
+        thermo_attribs["temperature-ranges"] = temperature_ranges
+        return thermo_attribs
+
+    def NASA9(self, thermo: etree.Element) -> Dict[str, Union[str, "THERMO_POLY_TYPE"]]:
+        """Process a NASA 9 thermo entry from XML to a dictionary."""
+        thermo_attribs = BlockMap({"model": "NASA9"})
+        data, temperature_ranges = self.process_polynomial(thermo, "NASA9")
+        thermo_attribs["data"] = data
+        thermo_attribs["temperature-ranges"] = temperature_ranges
+        thermo_attribs["temperature-ranges"] = FlowList(sorted(temperature_ranges))
+        return thermo_attribs
+
+    def const_cp(self, thermo: etree.Element) -> Dict[str, Union[str, float]]:
+        """Process a constant c_p thermo entry from XML to a dictionary."""
+        thermo_attribs = BlockMap({"model": "constant-cp"})
+        const_cp_node = thermo.find("const_cp")
+        if const_cp_node is None:
+            raise MissingXMLNode(
+                "The thermo node must constain a const_cp node", thermo
+            )
+        for node in const_cp_node:
+            thermo_attribs[node.tag] = get_float_or_units(node)
```

Comment:
The YAML parser doesn't seem to mind, see `ctml2yamlTest.test_diamond`. Should it be a case-sensitive comparison?

Reply:
Well, `diamond.xml` uses `t0 = 298.15` for all species, which is also the default value if the `T0` field is omitted. Here is where it's read -- I promise this is case sensitive:
https://github.com/Cantera/cantera/blob/ba330a797e9b9b3cecc84b3a9cb3f126d82015f1/src/thermo/SpeciesThermoFactory.cpp#L295-L303

Reply:
Oh, I see, the value for `t0` is just ignored, and since it's the default, it doesn't actually affect any of the results. OK, I had fixed this anyways, just in case 😄